### PR TITLE
refactor terraform to use modules, separate prod and staging

### PR DESCRIPTION
**Pull request by imjasonh**

### CODE REVIEW

**Comment by imjasonh** [comment]
This is applied to my project, and serving here: https://redirect.imjasonh.dev/chainguard/static:latest

```
$ crane manifest redirect.imjasonh.dev/chainguard/static:latest | jq '.mediaType'
"application/vnd.oci.image.index.v1+json"
```

If this LGTY I can start applying it to prod and then merge this when that's done.

**Comment by imjasonh** [comment]
I've `terraform import`ed most of the things in chainguard-images. Unfortunately there's still this diff:

```
  # module.gclb.google_certificate_manager_certificate.cert["cgr.dev"] must be replaced
-/+ resource "google_certificate_manager_certificate" "cert" {
      ~ id      = "projects/chainguard-images/locations/global/certificates/cgr-dev" -> (known after apply)
      - labels  = {} -> null
        name    = "cgr-dev"
      ~ project = "chainguard-images" -> (known after apply)

      ~ managed {
          ~ authorization_attempt_info = [
              - {
                  - details        = ""
                  - domain         = "cgr.dev"
                  - failure_reason = ""
                  - state          = "AUTHORIZED"
                },
            ] -> (known after apply)
          ~ dns_authorizations         = [ # forces replacement
              + "projects/chainguard-images/locations/global/dnsAuthorizations/cgr-dev",
            ]
          ~ provisioning_issue         = [] -> (known after apply)
          ~ state                      = "ACTIVE" -> (known after apply)
            # (1 unchanged attribute hidden)
        }

      - timeouts {}
    }
```

As I understand it, this means the cert will get deleted and re-created -- with the exact same DNS authorization we already have today. I think this will give us a few minutes of downtime while the cert provisions and DNS-authorizes.

I'm not sure if there's a better way to import this state so that TF doesn't think it has to delete and re-create the certs. 😒

**Comment by cpanato** [comment]
try to import using https://github.com/GoogleCloudPlatform/terraformer/blob/master/docs/gcp.md

**Comment by imjasonh** [comment]
I was able to get the diff to go away. I'm going to poke at this a bit more and roll it out before merging.

**Comment by imjasonh** [comment]
This is rolled out to prod now without any issue. 👍

File: terraform/modules/gclb/cert.tf
Comment (jdolitsky) [inline terraform/modules/gclb/cert.tf:47]: -2 ? this a blue/green thing?

**Review by jdolitsky — APPROVED** [review]

File: terraform/modules/gclb/cert.tf
Comment (imjasonh) [reply terraform/modules/gclb/cert.tf:47]: During our cert weirdness last week we added a primary cert map entry called `cgr-dev-2`, this just mirrors that into terraform.

If we want to we can remove the old non-primary one and promote this, I just didn't want to make prod diffs with this change.

**Pull request merged**
